vm/actor/src/builtin/chaos/mod.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod state;
+mod types;
+
+use super::{ACCOUNT_ACTOR_CODE_ID, SYSTEM_ACTOR_ADDR};
+use ipld_blockstore::BlockStore;
+use num_bigint::BigInt;
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+use runtime::{ActorCode, Runtime};
+pub use state::*;
+pub use types::*;
+use vm::{actor_error, ActorError, ExitCode, MethodNum, Serialized, METHOD_CONSTRUCTOR};
+// * Updated to specs-actors commit: 4784ddb8e54d53c118e63763e4efbcf0a419da28
+
+lazy_static! {
+    pub static ref CALLER_VALIDATION_BRANCH_NONE: BigInt = BigInt::from(0);
+    pub static ref CALLER_VALIDATION_BRANCH_TWICE: BigInt = BigInt::from(1);
+    pub static ref CALLER_VALIDATION_BRANCH_ADDR_NIL_SET: BigInt = BigInt::from(2);
+    pub static ref CALLER_VALIDATION_BRANCH_TYPE_NIL_SET: BigInt = BigInt::from(3);
+}
+
+/// Chaos actor methods available
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    CallerValidation = 2,
+    CreateActor = 3,
+}
+
+/// Chaos Actor
+pub struct Actor;
+
+impl Actor {
+    /// Constructor for Account actor
+    pub fn constructor<BS, RT>(_rt: &mut RT)
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        panic!("Constructor should not be called");
+    }
+
+    // CallerValidation violates VM call validation constraints.
+    //
+    //  CALLER_VALIDATION_BRANCH_NONE performs no validation.
+    //  CALLER_VALIDATION_BRANCH_TWICE validates twice.
+    //  CALLER_VALIDATION_BRANCH_ADDR_NIL_SET validates against an empty caller
+    //  address set.
+    //  CALLER_VALIDATION_BRANCH_TYPE_NIL_SET validates against an empty caller type set.
+    pub fn caller_validation<BS, RT>(rt: &mut RT, branch: BigInt) -> Result<(), ActorError>
```

Comment:
wrap the bigint in the `BigIntDe` wrapper structure we have (this is incorrect serialization format)

Alternatively you can just wrap the deserialization in the `invoke_method` call (which I'm thinking might be a better solution anyways)